src/Fmt_ast.ml

```diff
@@ -1749,6 +1749,25 @@ and fmt_expression c ?(box = true) ?pro ?epi ?eol ?parens ?(indent_wrap = 0)
                           )
                       $ fmt_if parens_bch (if imd then " )" else ")") )
                     $ fmt_if (not last) "@ "
+                | `K_R ->
```

Comment:
A lot of code could be factorized (eg. the `xcnd` part). What about using the `Params` pattern here ?

Reply:
Done, but not sure it's easier to read

Reply:
Not sure too. But I think it's a lot more maintainable now.